apps/web/src/components/Modal/AffiliateSunsetModal.tsx

```diff
@@ -0,0 +1,93 @@
+import { ReactNode, useCallback, useEffect } from 'react'
+
+import DisclaimerModal from 'components/DisclaimerModal'
+import { useUserIsInAffiliateListData } from 'hooks/useAffiliateSunsetList'
+import { useUserAcknowledgement } from 'hooks/useUserAcknowledgement'
+
+const transRegex = /(%[^%]+%)/
+
+function Trans({ text, data = {} }: { text: string; data?: { [key: string]: ReactNode } }) {
+  const parts = text.split(transRegex)
+  return parts.map((p) => {
+    if (!transRegex.test(p)) {
+      return p
+    }
+    const key = p.replace(/%/g, '')
+    return data[key] || p
+  })
+}
+
+export function AffiliateSunsetModal() {
+  const { t } = useTranslation()
+  const isInList = useUserIsInAffiliateListData()
+  const [ack, setACK] = useUserAcknowledgement('affiliate-referral-sunset-v2')
+  const onConfirm = useCallback(() => setACK(true), [setACK])
+
+  const [onOptionsConfirmModalPresent] = useModal(
+    <DisclaimerModal
+      bodyMaxWidth={['100%', '100%', '100%', '740px']}
+      bodyMaxHeight="80vh"
+      modalHeader={t('Important Update: Closure of the PancakeSwap Affiliate Program')}
+      header={
+        <>
+          <Text bold fontSize="1.25rem">
+            {t('Important Update')}
+          </Text>
+          <Text mt="1.5rem">
+            {t(`The PancakeSwap Affiliate Program will officially close on %time%`, {
+              time: 'December 31, 2024',
+            })}
+          </Text>
+          <Text mt="1.5rem">
+            <Trans
+              text={t(
+                `Please %claim% any pending trading discounts before this date, as discount redemptions will no longer be available after that.`,
+              )}
+              data={{
+                claim: (
+                  <LinkExternal
+                    bold
+                    style={{ display: 'inline-flex' }}
+                    showExternalIcon={false}
+                    href="/affiliates-program/dashboard"
+                  >
+                    {t('claim')}
+                  </LinkExternal>
+                ),
+              }}
+            />
+          </Text>
+          <Text mt="1.5rem">
+            {t(
+              'If you were referred by an affiliate who enabled commission sharing, please note that trading discounts from their referral link ended on November 25, 2024, and you will no longer receive discounts from this date onwards.',
+            )}
+          </Text>
+          <Text mt="1.5rem">
+            {t('Thank you for your understanding and being a valued PancakeSwap community member.')}
+          </Text>
+        </>
+      }
+      id="affiliate-sunset-modal"
+      checks={[
+        {
+          key: 'checkbox',
+          content: t(`I've read and understood the following update.`),
+        },
+      ]}
+      onSuccess={onConfirm}
+    />,
+    false,
+    false,
+    'affiliateSunsetModal',
+  )
+
+  useEffect(() => {
+    if (isInList && ack === false) {
+      onOptionsConfirmModalPresent()
```

Comment:
from an optimization point of view, how about moving the logic in useEffect for opening modal to a Callback passed to the hook that runs if data is true? Something like that would save us an effect. 

@Chef-Yogi

Reply:
I mean in the hook we already have `useUserIsInAffiliateListData` where we have data from the query